From: Bjorn Helgaas <bhelgaas@google.com>
Date: Tue, 5 Nov 2013 13:34:51 -0700
Subject: PCI: Drop warning about drivers that don't use pci_set_master()
Git-commit: fbeeb822f6f45cadf154d7b7cff1c13537cd799d
Patch-mainline: v3.13
References: http://lists.opensuse.org/opensuse-kernel/2013-10/msg00000.html

f41f064cf4 ("PCI: Workaround missing pci_set_master in pci drivers") made
pci_enable_bridge() turn on bus mastering if the driver hadn't done so
already.  It also added a warning in this case.  But there's no reason to
warn about it unless it's actually a problem to enable bus mastering here.

This patch drops the warning because I'm not aware of any such problem.

Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
CC: Paul Bolle <pebolle@tiscali.nl>
Acked-by: Jean Delvare <jdelvare@suse.de>

---
 drivers/pci/pci.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/pci/pci.c
+++ b/drivers/pci/pci.c
@@ -1156,10 +1156,8 @@ static void pci_enable_bridge(struct pci
 	pci_enable_bridge(dev->bus->self);
 
 	if (pci_is_enabled(dev)) {
-		if (!dev->is_busmaster) {
-			dev_warn(&dev->dev, "driver skip pci_set_master, fix it!\n");
+		if (!dev->is_busmaster)
 			pci_set_master(dev);
-		}
 		return;
 	}
 
